### ISSUE TITLE
In URLs, use unit index (aka string number), not the offset

### ISSUE THREAD

**Issue by iafan** [translate/pootle#4607]
To make URLs more user-friendly and understandable, expose unit index (the same number as shown in the top right unit navigation UI) in the URL, instead of the offset. If the unit has shifted within a N-unit result page, the string index parameter would be updated in the URL accordingly, indicating the change.

This might be seen as a cosmetic issue, but I think this can help users understand what these URLs are (not permalinks) and help developers/admins explain the logic behind URL reloading to the users.

**Comment by dwaynebailey** [comment]
@iafan just checking you mean the #N in the editor navbar and the #unit= in the URL should be adjusted the match?

My take on this is slightly different.  I'd prefer that we make the #N in the unit navbar a proper item that can be pushed to the clipboard and shared.  That way we change the behavior of how users would try to share links to units.  And the URL in the location bar is then not important anymore.

**Comment by iafan** [comment]
> just checking you mean the #N in the editor navbar and the #unit= in the URL should be adjusted the match?

No, this is about unit index (number) that is displayed in the next/previous unit navigation block and about `offset=X` to be replaced with something like `n=X` in the URL and have it adjusted when necessary based on the additionally provided value of `unit=NNNNNN`.

This issue is specifically about making *URLs* more understandable/useful, because people will always use URLs in addition to any permalink mechanism you're going to have. We already have unit permalinks, so I don't believe adding yet another kind of permalink will change user behavior wrt using or not using URLs. In the past we used to put some effort into making URLs user-friendly, and this, I believe, aligns with that effort.
